include/field.hxx

```diff
@@ -269,4 +271,42 @@ inline T filledFrom(const T& f, BoutReal fill_value) {
 template<typename T>
 T operator+(const T& f) {return f;}
 
+/// Check if all values of a field \p var are finite.  Loops over all points including the
+/// boundaries by default (can be changed using the \p rgn argument)
+/// If any element is not finite, throws an exception that includes the position of the
+/// first found.
+template<typename T>
+inline void checkFinite(const T& f, const std::string& name="field", const std::string& rgn="RGN_ALL") {
+  AUTO_TRACE();
+
```

Comment:
With #1730 this could include `static_assert(bout::utils::is_Field<T>::value, "checkFinite is only valid for Fields")`